adapters/beachfront/beachfront.go

```diff
@@ -24,41 +22,45 @@ const VideoEndpoint = "https://reachms.bfmio.com/bid.json?exchange_id="
 const VideoEndpointSuffix = "&prebidserver"
 
 const beachfrontAdapterName = "BF_PREBID_S2S"
-const beachfrontAdapterVersion = "0.3.0"
+const beachfrontAdapterVersion = "0.6.0"
+
+const DefaultVideoWidth = 300
+const DefaultVideoHeight = 250
 
 type BeachfrontAdapter struct {
 }
 
-type BeachfrontRequests struct {
-	Banner BeachfrontBannerRequest
-	Video  BeachfrontVideoRequest
+type beachfrontRequests struct {
+	Banner beachfrontBannerRequest
+	Video  []beachfrontVideoRequest
+	Audio  openrtb.Audio
+	Native openrtb.Native
 }
 
 // ---------------------------------------------------
 //              Video
 // ---------------------------------------------------
 
-type BeachfrontVideoRequest struct {
+type beachfrontVideoRequest struct {
```

Comment:
Below you marshal `beachfrontRequests.Video[i]` but `Video` is the `beachfrontVideoRequest` struct. Since this is private, the marshaling should fail. You might also want to review your tests cases that this failure was not already caught.

Reply:
I'm not clear on what should make this fail. I mean, yes, I set that struct to private figuring limiting scope to the minimum is generally good practice, and I saw no reason to make it public, not to say that there isn't one that I'm not getting, but it does actually work.  I am testing with Prebid.js test pages on a local npm install, which are sending a variety of requests, including a multi media / multiple imp request, to a running a local prebid server debug build in intellij. I can step through the debugger and see the marshaling work. Below, I have stepped through to i = 1. beachfrontRequests.Video[i] was marshaled to bytes the first time, then again the second time, with no error. The request gets appended to reqs, sent, received, processed, and I see the test ad video play on the test page.

![Screenshot from 2019-08-14 12-52-10](https://user-images.githubusercontent.com/5107143/63040069-a8eb3400-be92-11e9-85b8-fc11eab0501e.png)

Reply:
Hmm, perhaps it works since all the fields are public, even if the struct itself is private. The json package needs know what a `beachfrontVideoRequest` is so that it can generate the json representation of it for the payload you send to your servers. But if you have verified that the request is being generated properly despite the struct name being private, then I guess it is good.

Also yes, it is best in general to default to limiting scope. :)

Reply:
I'm going to go ahead and close this one and the one below, then. I'll be reading up on how the compiler handles this.  I suspect that this relates to why in go "private" and "public" are not actually the correct terms, but a package has "exported" identifiers and non-exported identifiers.  I don't quite get what the difference is besides semantics, but I suspect that this behavior is due to some specific detail of compilation that reflects that distinction. So off to RTFM.